ast/functions.go

```diff
@@ -407,7 +532,35 @@ type FuncCastExpr struct {
 
 // Restore implements Node interface.
 func (n *FuncCastExpr) Restore(ctx *RestoreCtx) error {
-	return errors.New("Not implemented")
+	switch n.FunctionType {
+	case CastFunction:
+		ctx.WriteKeyWord("CAST")
+		ctx.WritePlain("(")
```

Comment:
move the `WriteKeyWord` of function name outside the `switch`

Reply:
Here I may not be able to move the function name out of switch, because the function name of the statement is not saved in the expression.

Reply:
Okay.